### PR TITLE
Add ubuntu-xenial aarch64 Dockerfile

### PR DESCRIPTION
**Pull request by seemethere**
Signed-off-by: Eli Uriegas <eli.uriegas@docker.com>

### CODE REVIEW

**Comment by alexellis** [comment]
> We don't have official binary golang 1.7.5 tarballs for ARM64, eigher for Go or

FYI there is an official Go 1.9 beta binary available for ARM64 now. The project said that if it gets tested by the community they will continue with ARM64 binaries. Given the note above this could be useful for Docker CE etc.

**Comment by seemethere** [comment]
@alexellis, The new Go 1.9 beta binaries do build docker-ce successfully on aarch64 but we'll be compiling Go 1.8.3 from source until the core team can vet that Go 1.9 doesn't break any functionality.

**Comment by alexellis** [comment]
I know the Go version is 1.8.x The comment was as an FYI.

**Comment by rogaha** [comment]
@seemethere it seems that we need to add a dependency there: 

```
root@docker-aarch64:~/docker-ce-packaging/deb/debbuild/ubuntu-xenial# dpkg -i docker-ce_17.06.0~ce-0~ubuntu_arm64.deb
(Reading database ... 21956 files and directories currently installed.)
Preparing to unpack docker-ce_17.06.0~ce-0~ubuntu_arm64.deb ...
Unpacking docker-ce (17.06.0~ce-0~ubuntu) ...
dpkg: dependency problems prevent configuration of docker-ce:
 docker-ce depends on libltdl7 (>= 2.4.6); however:
  Package libltdl7 is not installed.

dpkg: error processing package docker-ce (--install):
 dependency problems - leaving unconfigured
Processing triggers for systemd (229-4ubuntu17) ...
Processing triggers for man-db (2.7.5-1) ...
Errors were encountered while processing:
 docker-ce
root@docker-aarch64:~/docker-ce-packaging/deb/debbuild/ubuntu-xenial# apt-get install libltdl7
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following NEW packages will be installed:
  libltdl7
0 upgraded, 1 newly installed, 0 to remove and 6 not upgraded.
1 not fully installed or removed.
Need to get 35.5 kB of archives.
After this operation, 80.9 kB of additional disk space will be used.
Get:1 http://ports.ubuntu.com/ubuntu-ports xenial/main arm64 libltdl7 arm64 2.4.6-0.1 [35.5 kB]
Fetched 35.5 kB in 0s (1,010 kB/s)
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 1.)
debconf: falling back to frontend: Readline
Selecting previously unselected package libltdl7:arm64.
(Reading database ... 22165 files and directories currently installed.)
Preparing to unpack .../libltdl7_2.4.6-0.1_arm64.deb ...
Unpacking libltdl7:arm64 (2.4.6-0.1) ...
Processing triggers for libc-bin (2.23-0ubuntu9) ...
Setting up libltdl7:arm64 (2.4.6-0.1) ...
Setting up docker-ce (17.06.0~ce-0~ubuntu) ...
Installing new version of config file /etc/default/docker ...
Installing new version of config file /etc/init.d/docker ...
Installing new version of config file /etc/init/docker.conf ...
Processing triggers for libc-bin (2.23-0ubuntu9) ...
Processing triggers for systemd (229-4ubuntu17) ...
root@docker-aarch64:~/docker-ce-packaging/deb/debbuild/ubuntu-xenial# dpkg -i docker-ce_17.06.0~ce-0~ubuntu_arm64.deb
(Reading database ... 22172 files and directories currently installed.)
Preparing to unpack docker-ce_17.06.0~ce-0~ubuntu_arm64.deb ...
Unpacking docker-ce (17.06.0~ce-0~ubuntu) over (17.06.0~ce-0~ubuntu) ...
Setting up docker-ce (17.06.0~ce-0~ubuntu) ...
Processing triggers for systemd (229-4ubuntu17) ...
Processing triggers for man-db (2.7.5-1) ...
root@docker-aarch64:~/docker-ce-packaging/deb/debbuild/ubuntu-xenial#
root@docker-aarch64:~/docker-ce-packaging/deb/debbuild/ubuntu-xenial# docker version
Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Thu Jul  6 08:51:38 2017
 OS/Arch:      linux/arm64

Server:
 Version:      17.06.0-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Thu Jul  6 08:50:12 2017
 OS/Arch:      linux/arm64
 Experimental: false
```

File: deb/ubuntu-xenial/Dockerfile.aarch64 [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# We don't use the official ARMv6 released binaries as a GOROOT_BOOTSTRAP, because
+# not all ARM64 platforms support 32-bit mode. 32-bit mode is optional for ARMv8.
+ENV GO_VERSION 1.8.3
+RUN mkdir /usr/src/go && curl -fsSL https://golang.org/dl/go${GO_VERSION}.src.tar.gz | tar -v -C /usr/src/go -xz --strip-components=1 \
+	&& cd /usr/src/go/src \
+	&& GOOS=linux GOARCH=arm64 GOROOT_BOOTSTRAP="$(go env GOROOT)" ./make.bash
+
+ENV GOPATH /go
+ENV PATH /go/bin:/usr/src/go/bin:$PATH
+ENV DOCKER_BUILDTAGS apparmor pkcs11 seccomp selinux
```
Comment (andrewhsu) [inline deb/ubuntu-xenial/Dockerfile.aarch64]: no need for pkcs11, see https://github.com/moby/moby/commit/b877fc31c51baaf1ec42501ab366dd05a5b6bb21#diff-0a5836d551d7f223bdc8b46372ca0089

Comment (seemethere) [inline deb/ubuntu-xenial/Dockerfile.aarch64]: Updated and force pushed, plus rebased off of `master`

**Comment by andrewhsu** [comment]
LGTM

**Pull request merged**